packages/react/src/select/trigger/useSelectTrigger.ts

```diff
@@ -85,12 +93,20 @@ export function useSelectTrigger(
       tabIndex: disabled ? -1 : 0,
       ref: handleRef,
       onFocus(event) {
-        setTypeaheadReady(true);
         setFocused(true);
         // The popup element shouldn't obscure the focused trigger.
         if (open && alignItemWithTriggerActiveRef.current) {
           setOpen(false, event.nativeEvent, 'focus-out');
         }
+
+        // Saves a re-render on initial click: `typeaheadReady === true` mounts
+        // the items before `open === true`. We could sync those cycles better
+        // without a timeout, but this is enough for now.
+        //
+        // XXX: might be causing `act()` warnings.
+        timeoutFocus.start(0, () => {
+          store.set('typeaheadReady', true);
+        });
```

Comment:
With a large select, this is much better as focus immediately gets set on the element. Previously it was blocked synchronously